CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSVReader.java

```diff
@@ -8,107 +8,88 @@
+    static String setFieldName(int i, Map<String,String>schema)
     {
-        String field = String.format("field%d",i); 
-        if (schema != null)
-        {
-            if (schema.get(field)!= null)
-            {
+        String field = String.format("field%d", i); 
+        if (schema != null){
+            if (schema.get(field)!= null){
                 field = schema.get(field) ; 
             }
         }
-
         return field ; 
     }
-    @SuppressWarnings("unchecked")
-    static public ArrayList<Map<String,String>> execute(String csvFile,  Map<String, Object>     config,   ExtensionWebSocketClient oClient  ) {
 
+    @SuppressWarnings("unchecked")
+    static public ArrayList<Map<String,String>> execute(String csvFile, Map<String, Object> config, ExtensionWebSocketClient oClient) {
         String line = "";
         int numOfRecords ; 
-        int numOfProcessedLinesInSegment = 0 ; 
+        int numOfProcessedLinesFromFile = 0 ; 
         Map<String,String> schema = null;
 
-        if (config.get("schema") != null)
-        {
+        if (config.get("schema") != null){
             schema = (Map<String,String>) config.get("schema");
         }
 
-        String delimiter = ";";
-        if (config.get("delimiter") != null)
-        {
+        String delimiter = ",";
+        if (config.get("delimiter") != null){
             delimiter = config.get("delimiter").toString();
         }
 
-
         int numLinesInEvent = (int) config.get(MAX_LINES_IN_EVENT);
-
         ArrayList<Map<String,String>> file = new ArrayList<Map<String,String>>(); 
 
         try (BufferedReader br = new BufferedReader(new FileReader(csvFile))) {
-
             numOfRecords = 0 ; 
             while ((line = br.readLine()) != null) {
-                numOfProcessedLinesInSegment++;
+                numOfProcessedLinesFromFile++;
                 // use comma as separator
                 String[] values = line.split(delimiter);
-
                 Map<String, String> lineValues = new HashMap<String, String>(); 
                 
-                for (int i = 0 ; i < values.length ; i++)
-                {
-                    String currField = setFieldName(i,schema);
-                    lineValues.put(currField, values[i]);
+                int schemaFieldIndex = 0 ; 
+                for (int i = 0 ; i < values.length ; i++){
+                    if (values[i].length() != 0){
+                        String currField = setFieldName(schemaFieldIndex,schema);
+                        lineValues.put(currField, values[i]);
+                        schemaFieldIndex++;
+                    }
                 }
 
                 file.add(lineValues);
                 numOfRecords++ ; 
 
-                if (numOfRecords >= numLinesInEvent)
-                {
+                if (numOfRecords >= numLinesInEvent){
                     //TODO : this is for testing logic. could have been smarter...
                     if (oClient != null) 
```

Comment:
If statements are ALWAYS of the form

```
    if (...) {
        // then body
    } else {
        // else body
    }
```

If there's no else stuff to do, then leave that out.  But we ALWAYS use the brace form.  Never the _only one statement_ form.

Reply:
ok , done